Error in ecs_world! macro

Tried the example in the readme on a fresh project and I immediately get an error in the `ecs_world!` call

```rust
use gecs::prelude::*;

// Components -- these must be pub because the world is exported as pub as well.
pub struct CompA(pub u32);
pub struct CompB(pub u32);
pub struct CompC(pub u32);

ecs_world! {
    // Declare two archetypes, ArchFoo and ArchBar.
    ecs_archetype!(ArchFoo, 100, CompA, CompB); // Fixed capacity of 100 entities.
    ecs_archetype!(ArchBar, dyn, CompA, CompC); // Dynamic (dyn) entity capacity.
}

fn main() {
    let mut world = World::default(); // Initialize an empty new ECS world.

    // Add entities to the world by populating their components and receive their handles.
    let entity_a = world.create::<ArchFoo>((CompA(1), CompB(20)));
    let entity_b = world.create::<ArchBar>((CompA(3), CompC(40)));

    // Each archetype now has one entity.
    assert_eq!(world.len::<ArchFoo>(), 1);
    assert_eq!(world.len::<ArchBar>(), 1);

    // Look up each entity and check its CompB or CompC value.
    assert!(ecs_find!(world, entity_a, |c: &CompB| assert_eq!(c.0, 20)));
    assert!(ecs_find!(world, entity_b, |c: &CompC| assert_eq!(c.0, 40)));

    // Add to entity_a's CompA value.
    ecs_find!(world, entity_a, |c: &mut CompA| { c.0 += 1; });

    // Sum both entities' CompA values with one iter despite being different archetypes.
    let mut sum = 0;
    ecs_iter!(world, |c: &CompA| { sum += c.0 });
    assert_eq!(sum, 5); // Adding 2 + 3 -- recall that we added 1 to entity_a's CompA.

    // Collect both entities that have a CompA component.
    let mut found = Vec::new();
    ecs_iter!(world, |entity: &EntityAny, _: &CompA| { found.push(*entity); });
    assert!(found == vec![entity_a.into(), entity_b.into()]);

    // Destroy both entities -- this will return an Option containing their components.
    assert!(world.destroy(entity_a).is_some());
    assert!(world.destroy(entity_b).is_some());

    // Try to look up a stale entity handle -- this will return false.
    assert_eq!(ecs_find!(world, entity_a, |_: &Entity<ArchFoo>| { panic!() }), false);
}
```

cargo build output:
```
Compiling dispatch-test v0.1.0 (/home/do/Projects/rust/dispatch-test)
error[E0599]: no function or associated item named `default` found for trait `gecs::traits::World`
  --> src/main.rs:15:28
   |
15 |     let mut world = World::default(); // Initialize an empty new ECS world.
   |                            ^^^^^^^ function or associated item not found in `World`

error[E0308]: mismatched types
  --> src/main.rs:8:1
   |
8  | / ecs_world! {
9  | |     // Declare two archetypes, ArchFoo and ArchBar.
10 | |     ecs_archetype!(ArchFoo, 100, CompA, CompB); // Fixed capacity of 100 entities.
11 | |     ecs_archetype!(ArchBar, dyn, CompA, CompC); // Dynamic (dyn) entity capacity.
12 | | }
   | |_^ expected `bool`, found `Option<()>`
...
26 |       assert!(ecs_find!(world, entity_a, |c: &CompB| assert_eq!(c.0, 20)));
   |               ----------------------------------------------------------- in this macro invocation
   |
   = note: expected type `bool`
              found enum `Option<()>`
   = note: this error originates in the macro `::gecs::__internal::__ecs_find` which comes from the expansion of the macro `ecs_find` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `Option::is_some` to test if the `Option` has a value
   |
12 | }.is_some()
   |  ++++++++++

error[E0308]: mismatched types
  --> src/main.rs:8:1
   |
8  | / ecs_world! {
9  | |     // Declare two archetypes, ArchFoo and ArchBar.
10 | |     ecs_archetype!(ArchFoo, 100, CompA, CompB); // Fixed capacity of 100 entities.
11 | |     ecs_archetype!(ArchBar, dyn, CompA, CompC); // Dynamic (dyn) entity capacity.
12 | | }
   | |_^ expected `bool`, found `Option<_>`
...
26 |       assert!(ecs_find!(world, entity_a, |c: &CompB| assert_eq!(c.0, 20)));
   |               ----------------------------------------------------------- in this macro invocation
   |
   = note: expected type `bool`
              found enum `Option<_>`
   = note: this error originates in the macro `::gecs::__internal::__ecs_find` which comes from the expansion of the macro `ecs_find` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `Option::is_some` to test if the `Option` has a value
   |
12 | }.is_some()
   |  ++++++++++

error[E0308]: mismatched types
  --> src/main.rs:8:1
   |
8  | / ecs_world! {
9  | |     // Declare two archetypes, ArchFoo and ArchBar.
10 | |     ecs_archetype!(ArchFoo, 100, CompA, CompB); // Fixed capacity of 100 entities.
11 | |     ecs_archetype!(ArchBar, dyn, CompA, CompC); // Dynamic (dyn) entity capacity.
12 | | }
   | |_^ expected `bool`, found `Option<()>`
...
27 |       assert!(ecs_find!(world, entity_b, |c: &CompC| assert_eq!(c.0, 40)));
   |               ----------------------------------------------------------- in this macro invocation
   |
   = note: expected type `bool`
              found enum `Option<()>`
   = note: this error originates in the macro `::gecs::__internal::__ecs_find` which comes from the expansion of the macro `ecs_find` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `Option::is_some` to test if the `Option` has a value
   |
12 | }.is_some()
   |  ++++++++++

error[E0308]: mismatched types
  --> src/main.rs:8:1
   |
8  | / ecs_world! {
9  | |     // Declare two archetypes, ArchFoo and ArchBar.
10 | |     ecs_archetype!(ArchFoo, 100, CompA, CompB); // Fixed capacity of 100 entities.
11 | |     ecs_archetype!(ArchBar, dyn, CompA, CompC); // Dynamic (dyn) entity capacity.
12 | | }
   | |_^ expected `bool`, found `Option<_>`
...
27 |       assert!(ecs_find!(world, entity_b, |c: &CompC| assert_eq!(c.0, 40)));
   |               ----------------------------------------------------------- in this macro invocation
   |
   = note: expected type `bool`
              found enum `Option<_>`
   = note: this error originates in the macro `::gecs::__internal::__ecs_find` which comes from the expansion of the macro `ecs_find` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `Option::is_some` to test if the `Option` has a value
   |
12 | }.is_some()
   |  ++++++++++

error[E0308]: mismatched types
  --> src/main.rs:47:5
   |
47 |     assert_eq!(ecs_find!(world, entity_a, |_: &Entity<ArchFoo>| { panic!() }), false);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Option<_>`, found `bool`
   |
   = note: expected enum `Option<_>`
              found type `bool`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try wrapping the expression in `Some`
  --> /home/do/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs:40:35
   |
40 |                 if !(*left_val == Some(*right_val)) {
   |                                   +++++          +

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `dispatch-test` (bin "dispatch-test") due to 6 previous errors
```

Rustup info:
```
stable-x86_64-unknown-linux-gnu
rustc 1.71.1 (eb26296b5 2023-08-03)
```

Edition 2021. Using gecs 0.3.0

--- a/README.md
+++ b/README.md
@@ -53,19 +53,20 @@ ecs_world! {
 }
 
 fn main() {
-    let mut world = World::default(); // Initialize an empty new ECS world.
+    let mut world = EcsWorld::default(); // Initialize an empty new ECS world.
 
     // Add entities to the world by populating their components and receive their handles.
     let entity_a = world.create::<ArchFoo>((CompA(1), CompB(20)));
     let entity_b = world.create::<ArchBar>((CompA(3), CompC(40)));
 
     // Each archetype now has one entity.
-    assert_eq!(world.len::<ArchFoo>(), 1);
-    assert_eq!(world.len::<ArchBar>(), 1);
+    assert_eq!(world.archetype::<ArchFoo>().len(), 1);
+    assert_eq!(world.archetype::<ArchBar>().len(), 1);
 
     // Look up each entity and check its CompB or CompC value.
-    assert!(ecs_find!(world, entity_a, |c: &CompB| assert_eq!(c.0, 20)));
-    assert!(ecs_find!(world, entity_b, |c: &CompC| assert_eq!(c.0, 40)));
+    // We use the is_some() check here to make sure the entity was indeed found.
+    assert!(ecs_find!(world, entity_a, |c: &CompB| assert_eq!(c.0, 20)).is_some());
+    assert!(ecs_find!(world, entity_b, |c: &CompC| assert_eq!(c.0, 40)).is_some());
 
     // Add to entity_a's CompA value.
     ecs_find!(world, entity_a, |c: &mut CompA| { c.0 += 1; });
@@ -84,8 +85,8 @@ fn main() {
     assert!(world.destroy(entity_a).is_some());
     assert!(world.destroy(entity_b).is_some());
 
-    // Try to look up a stale entity handle -- this will return false.
-    assert_eq!(ecs_find!(world, entity_a, |_: &Entity<ArchFoo>| { panic!() }), false);
+    // Try to look up a stale entity handle -- this will return None.
+    assert!(ecs_find!(world, entity_a, |_: &Entity<ArchFoo>| { panic!() }).is_none());
 }
 ```
 License

